botcity/core/bot.py

```diff
@@ -443,6 +446,44 @@ def find_text(self, label, x=None, y=None, width=None, height=None, *, threshold
         return self.find_until(label, x, y, width, height, threshold=threshold, matching=matching,
                                waiting_time=waiting_time, best=best, grayscale=True)
 
+    def find_process(self, name: str = None, pid: str = None) -> Generator:
+        """
+        Find a process by name or PID
+
+        Args:
+            name (str): The process name.
+            pid (str) or (int): The PID (Process Identifier).
+
+        Return:
+            generator (Generator): The Generator Object.
+        """
+        for process in psutil.process_iter():
+            try:
+                if (name is not None and process.name() == name) or \
+                        (pid is not None and process.pid == pid):
```

Comment:
For `name` we could do a partial match checking if `name` is part of `process.name()`.

Here is a suggestion:
```suggestion
                if (name is not None and name in process.name()) or \
                        (pid is not None and process.pid == pid):
```

This will allow us to search for a process without knowing its full name or have an exact match.